src/Domain/MigrationStep/s150_ProductVariationMigration/MixedVariation/MixedVariationProductMigrator.php

```diff
@@ -0,0 +1,94 @@
+    private $productRepository;
+
+    /** @var ProductModelRepository */
+    private $productModelRepository;
+
+    /** @var InnerVariationProductVariantTransformer */
+    private $innerVariationProductVariantTransformer;
+
+    /** @var VariantGroupProductVariantTransformer */
+    private $variantGroupProductVariantTransformer;
+
+    /** @var ProductModelSaver */
+    private $productModelSaver;
+
+    public function __construct(
+        ProductModelBuilder $productModelBuilder,
+        ProductRepository $productRepository,
+        ProductModelRepository $productModelRepository,
+        ProductModelSaver $productModelSaver,
+        InnerVariationProductVariantTransformer $innerVariationProductVariantTransformer,
+        VariantGroupProductVariantTransformer $variantGroupProductVariantTransformer
+    )
+    {
+        $this->productModelBuilder = $productModelBuilder;
+        $this->productRepository = $productRepository;
+        $this->productModelRepository = $productModelRepository;
+        $this->innerVariationProductVariantTransformer = $innerVariationProductVariantTransformer;
+        $this->variantGroupProductVariantTransformer = $variantGroupProductVariantTransformer;
+        $this->productModelSaver = $productModelSaver;
+    }
+
+    /**
+     * Migrates products variations by creating products models and transforming products in product variants.
+     *  - A root product model is created for each variant group of a family an variant axes combination.
+     *  - A sub product model is created for each product grouped in the variant group, having variation via an inner-variation-type.
+     *  - Each product variation of the product grouped in the variant group is transformed to a product variant having the sub product model as parent.
+     *  - Each product grouped in the variant group that doesn't have variation is transformed to a product variant having the root product model as parent.
+     */
+    public function migrateProducts(MixedVariation $mixedVariation, FamilyVariant $familyVariant, DestinationPim $pim): void
+    {
+        $variantGroupCombination = $mixedVariation->getVariantGroupCombination();
+        $innerVariationType = $mixedVariation->getInnerVariationType();
+
+        foreach ($variantGroupCombination->getGroups() as $variantGroupCode) {
+            $rootProductModel = $this->productModelBuilder->buildRootProductModel($variantGroupCode, $familyVariant, $pim);
+            $rootProductModel = $this->productModelRepository->persist($rootProductModel, $pim);
+
+            $productsHavingVariants = $mixedVariation->getVariantGroupProductsHavingVariants($variantGroupCode);
+
+            foreach ($productsHavingVariants as $parentProduct) {
+                $subProductModel = $this->productModelBuilder->buildSubProductModel($rootProductModel, $parentProduct, $familyVariant, $pim);
+                $subProductModel = $this->productModelSaver->save($subProductModel, $pim);
```

Comment:
That will be replaced by `productModelRepository->persist` in a next PR where the repository will use the API to persist. I can't use the import without doing useless work to re-format the attributes values.